src/shogun/regression/KrrNystrom.cpp

```diff
@@ -0,0 +1,79 @@
+
+/*
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Written (W) 2016 Fredrik Hallgren
+ * Copyright (C) 1999-2009 Fraunhofer Institute FIRST and Max-Planck-Society
+ */
+
+#include <shogun/regression/KrrNystrom.h>
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/mathematics/eigen3.h>
+#include <shogun/mathematics/Math.h>
+
+using namespace shogun;
+using namespace Eigen;
+
+CKrrNystrom::CKrrNystrom() : CKernelRidgeRegression()
+{
+	init();
+}
+
+CKrrNystrom::CKrrNystrom(float64_t tau, int32_t m, CKernel* k, CLabels* lab)
+: CKernelRidgeRegression(tau, k, lab)
+{
+	init();
+
+	m_m=m;
+}
+
+void CKrrNystrom::init()
+{
+	// Initialize parameters
+	m_m=1000;  // TODO change to what seems to make sense
+	// TODO check that less than n
+}
+
+bool CKrrNystrom::solve_krr_system()
+{
+	SGMatrix<float64_t> kernel_matrix(kernel->get_kernel_matrix());
```

Comment:
For Nystrom, the point is *not* to compute the full kernel matrix, but only the subset you subsample below. Otherwise there is no gain in memory usage